<a href="https://colab.research.google.com/github/joseydyrueda-code/programacion1/blob/main/parcial2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>


#Joseydy Andrea Perozo Rueda
*   Pregrado Matematicas
*   2260665



**Instrucción 1**: Genere un arreglo de NumPy con 50 números enteros aleatorios entre 1XY y 19Z,
donde XY corresponde a los dos últimos dígitos de la TI o CC y Z al último dígito de la edad.
Utilice el siguiente código:


In [ ]:
import pandas as pd
import numpy as np
np.random.seed(12)
datos = np.random.randint(145, 194, 50)
df = pd.DataFrame({
         "datos": datos
         })



In [ ]:
datos

array([156, 172, 151, 147, 148, 148, 157, 193, 167, 150, 158, 170, 179,
       156, 155, 145, 185, 157, 158, 163, 170, 180, 181, 180, 178, 175,
       177, 163, 167, 161, 165, 151, 190, 154, 190, 175, 193, 177, 191,
       172, 170, 169, 160, 170, 178, 149, 177, 149, 146, 188])

1. Implemente una filtración sobre el elemento datos e imprima el elemento con los valores
resultantes.


In [ ]:
df.loc[df["datos"] > 180]

,datos
7,193
16,185
22,181
32,190
34,190
36,193
38,191
49,188


2. Escale cada uno de los valores de datos utilizando una función lambda

In [ ]:

import pandas as pd
x = [1, 2, 3, 4]
x = pd.DataFrame(x)
normalizar = lambda x: (x - x.mean()) / x.std()
resultado = normalizar(x)
print(resultado)

          0
0 -1.161895
1 -0.387298
2  0.387298
3  1.161895


3. Implemente el método .describe() e interprete los resultados.


In [ ]:
df.describe()

,0
count,50.000000
mean,155.360000
std,25.339942
min,121.000000
25%,133.000000
50%,154.500000
75%,178.750000
max,195.000000





*   
*   el total de elementos es 50
*   la desviacion estandar es de 25.3
*   el valor maximo es 195
*   el valor minimo es 121
*   el promedio es 155




4. Convierta el elemento datos en un DataFrame usando pandas. Imprima el elemento.

In [ ]:

df

,datos
0,156
1,172
2,151
3,147
4,148
5,148
6,157
7,193
8,167
9,150


5. Exporte los resultados a un archivo CSV llamado resultados.csv

In [152]:
x.to_csv("x.csv")

In [157]:
from google.colab import files
import csv
resultados_csv = [
([156, 172, 151, 147, 148, 148, 157, 193, 167, 150, 158, 170, 179,
       156, 155, 145, 185, 157, 158, 163, 170, 180, 181, 180, 178, 175,
       177, 163, 167, 161, 165, 151, 190, 154, 190, 175, 193, 177, 191,
       172, 170, 169, 160, 170, 178, 149, 177, 149, 146, 188])
]

with open("datos.csv", "w", newline="") as f:
    writer = csv.writer(f)
    writer.writerows(resultados_csv)

files.download("datos.csv")


<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

Instrucción 2: Construya un DataFrame en pandas con información de su elección. El
DataFrame debe cumplir con: (1) al menos 4 columnas (por ejemplo: nombre, edad, ingreso,
ciudad), (2) al menos 5 filas (registros) y (3) incluir datos numéricos y categóricos (o variable
cualitativa)


In [137]:
import pandas as pd

df = pd.DataFrame({
    "nombre": ["Juan", "Ana", "Jose", "Arturo", "Sofia"],
    "edad": [25, 18, 43, 27, 36],
    "ocupacion": ["Profesor", "Estudiante", "Doctor", "Taxista", "Odontologa"],
    "ciudad": ["Bogota", "Medellin", "Cali", "Barranquilla" "Bucaramanga"]
  })
df

ValueError: All arrays must be of the same length

#parte 2 del parcial


Proyecto: Una institución educativa busca analizar el desempeño académico de un mismo
grupo de estudiantes en dos evaluaciones distintas. Se plantea la posibilidad de que existan
diferencias en las calificaciones promedio entre ambas pruebas. Con este propósito, se requiere
desarrollar un paquete que permita realizar la comparación de los resultados para cualquier
conjunto de estudiantes.

1. Genere dos muestras aleatorias de tamaño 5W, donde W corresponden al último dígito del
día de cumpleaños:

In [154]:
import numpy as np
import pandas as pd

np.random.seed(1)

evaluacion_A = np.random.normal(loc=7.2, scale=0.4, size=57)
evaluacion_B = np.random.gamma(shape=306, scale=0.0229, size=57)

df = pd.DataFrame({
    "evaluacion 1": evaluacion_A,
    "evaluacion 2": evaluacion_B
})
df

,evaluacion 1,evaluacion 2
0,7.849738,7.379207
1,6.955297,7.561472
2,6.988731,7.272981
3,6.770813,6.933277
4,7.546163,7.373442
5,6.279385,6.653136
6,7.897925,7.111712
7,6.895517,6.298543
8,7.327616,7.637774
9,7.100252,6.398590


2. Construya un DataFrame en pandas que contenga ambas muestras.

In [158]:
df.describe()

,evaluacion 1,evaluacion 2
count,57.000000,57.000000
mean,7.188751,7.105223
std,0.376927,0.373595
min,6.279385,6.298543
25%,6.925131,6.873962
50%,7.123266,7.084545
75%,7.434649,7.277549
max,8.040102,8.037917


3. Realice un análisis descriptivo para cada una de las pruebas: Media, Mediana, máximo,
mínimo, desviación estándar, cuantil 25% y cuantil 75%.



*   desviacion estandar de 0.37
*   el minimo es de 6.27
*   el maximo es de 8.04
*   el promedio es de 7.1
*   el 25% de la evaluacion uno tiene 6.9 y el de la evaluacion dos tiene 6.8
*   el 75% de la evaluacion uno tiene 7.4 y de la evaluacion dos 7.2





Importe el siguiente módulo estadístico:
from scipy import stats

In [159]:
from scipy import stats

Aplique prueba de normalidad de Shapiro-Wilk para cada prueba ¿se cumple el supuesto
de normalidad con una confianza del 95%?

In [164]:
from scipy.stats import shapiro
shapiro(df["evaluacion 1" "evaluacion 2"])

KeyError: 'evaluacion 1evaluacion 2'

#